### PR TITLE
Merge release 0.24 to main

### PR DESCRIPTION
**Pull request by squat**
This commit follows our release process and merges the latest commits/tags from the release-0.24 branch back into main. Let's merge this after https://github.com/thanos-io/thanos/pull/5017. This will likely need another rebase.

cc @GiedriusS

### CODE REVIEW

**Comment by squat** [comment]
ping @GiedriusS
same issue as before. do you mind if I set DCO to `pass` for your commits?

File: VERSION [hunk 1]
```diff
@@ -1 +1 @@
-0.24.0-dev
+0.24.0
```
Comment (GiedriusS) [inline VERSION:1]: I think this now needs to be `0.25.0-dev`?

**Review by GiedriusS — DISMISSED** [review]
:+1: yes, feel free to do that. Thx for this!

File: VERSION (hunk 1 above)
Comment (squat) [reply VERSION:1]: Ach yes that's right

Comment (squat) [reply VERSION]: :tada: done

**Review by GiedriusS — APPROVED** [review]

**Pull request merged**